adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -60,31 +61,35 @@
     private BrokerResumeResultReceiver mBrokerResumeResultReceiver = null;
     private static final int TIMEOUT_FOR_BROKER_RESULT = 10 * 60 * 1000;
 
-    /*Used for silent request telemetry data logging.*/
-    private boolean mAcquireTokenSilentWithBroker = false;
-
     /**
      * Instance validation related calls are serviced inside Discovery as a
      * module.
      */
     private Discovery mDiscovery = new Discovery();
 
     /**
+     * Event Variable for the acquireToken API called. This will track whether the API succeeded or not.
+     */
+    private APIEvent mAPIEvent;
+
+    /**
      * Constructor for {@link AcquireTokenRequest}.
      */
-    AcquireTokenRequest(final Context appContext, final AuthenticationContext authContext) {
+    AcquireTokenRequest(final Context appContext, final AuthenticationContext authContext, final APIEvent apiEvent) {
         mContext = appContext;
         mAuthContext = authContext;
 
-        if (authContext.getCache() != null) {
+        if (authContext.getCache() != null && apiEvent != null) {
             mTokenCacheAccessor = new TokenCacheAccessor(authContext.getCache(),
-                    authContext.getAuthority());
+                    authContext.getAuthority(), apiEvent.getRequestId());
```

Comment:
the request id is telemetry request id, right? if so, can we say telemetryRequestId just to make it clear. AuthenticationRequest is already having a request id, don't make it confusing. :)